src/robothub/replay_camera.py

```diff
@@ -0,0 +1,542 @@
+            img_frame.setHeight(height)
+            return img_frame
+
+        def get_next_frame():
+            if not self._cap.isOpened():
+                return None
+
+            frame = None
+            for _ in range(2):
+                ok, frame = self._cap.read()
+                if not ok and self._loop:
+                    self._cap.release()
+                    if self._path_type == PathType.VIDEO:
+                        self._cap = cv2.VideoCapture(str(self._path))
+                    else:
+                        self._cap.reset()  # type:ignore
+                    continue
+                break
+
+            return frame
+
+        while True:
+            start = time.monotonic()
+
+            # NOTE(miha): Returned frame is in BGR format
+            frame = get_next_frame()
+            if frame is None:
+                break
+
+            nv12_frame = BGR2YUV_NV12(frame)
+            preview_frame = cv2.resize(frame, (self._preview_width, self._preview_height))
+
+            timestamp: timedelta = timedelta(seconds=time.time() - send_video_frames_start)
+
+            video_img_frame = create_img_frame(
+                data=nv12_frame,
+                width=self._video_width,
+                height=self._video_height,
+                type=dai.ImgFrame.Type.NV12,
+                sequence_number=sequence_number,
+                timestamp=timestamp,
+            )
+            preview_img_frame = create_img_frame(
+                data=to_planar(preview_frame, (self._preview_width, self._preview_height)),
+                width=self._preview_width,
+                height=self._preview_height,
+                type=dai.ImgFrame.Type.BGR888p,
+                sequence_number=sequence_number,
+                timestamp=timestamp,
+            )
+
+            sequence_number += 1
+
+            # TODO(miha): Send to other queues as well
+            self._video_queue.send(video_img_frame)
+            self._preview_queue.send(preview_img_frame)
+
+            # NOTE(miha): Mock input control commands
+            if self._input_control_queue.has():
+                ctrl: dai.CameraControl = self._input_control_queue.get()  # type: ignore
+
+                # TODO(miha): Send image to still queue
+                if ctrl.getCaptureStill():
+                    pass
+
+            process_time = time.monotonic() - start
+            if process_time > (1000.0 / self._fps) / 1000.0:
+                logging.error(
+                    f"Proccessing time ({process_time}) didn't hit the set camera FPS deadline ({1000.0 / self._fps})"
+                )
+            time_to_sleep = max((1000.0 / self._fps / 1000.0) - process_time, 0)
+            logging.debug(f"process_time: {process_time}, time_to_sleep: {time_to_sleep}")
+            time.sleep(time_to_sleep)
+
+        time.sleep(self._fps)  # NOTE(miha): Wait for last message to arrive
+        self._cap.release()
+        cv2.destroyAllWindows()
```

Comment:
why is this line needed?

Reply:
It is not, the user should handle cv2 window.